TrackletAlgorithm/MatchProcessor.h

```diff
@@ -431,13 +1081,36 @@ void MatchCalculator(BXType bx,
   }
   
   // For first tracklet, pick up the phi cut value
+  best_delta_z = (newtracklet)? LUT_matchcut_z[proj_seed] : best_delta_z;
   best_delta_phi = (newtracklet)? LUT_matchcut_phi[proj_seed] : best_delta_phi;
+  if(newtracklet) {
+    if(isPSStub) {
+      best_delta_rphi = LUT_matchcut_PSrphi[proj_seed];
+      best_delta_r = LUT_matchcut_PSr[proj_seed];
+    }
+    else  {
+      best_delta_rphi = LUT_matchcut_2Srphi[proj_seed];
+      best_delta_r = LUT_matchcut_2Sr[proj_seed];
+    }
+  }
 
   // Check that matches fall within the selection window of the projection 
-  if ((delta_z_fact < LUT_matchcut_z[proj_seed]) && (delta_z_fact >= -LUT_matchcut_z[proj_seed]) && (abs_delta_phi <= best_delta_phi)){
+  bool barrel_match = (delta_z_fact < best_delta_z) && (delta_z_fact >= -best_delta_z) && (abs_delta_phi < best_delta_phi);
+  bool disk_match = isPSStub ? ((abs_delta_phi * ap_uint<12>(stub_ps_r)) < best_delta_rphi) && (abs_delta_r < best_delta_r) : ((abs_delta_phi * ap_uint<12>(tmp_stubr)) < best_delta_rphi) && (abs_delta_r < best_delta_r);
+  disk_match = isMatch ? disk_match && (abs_delta_phi < best_delta_phi) : disk_match;
+  isMatch |= disk_match;
```

Comment:
isMatch only seems to be updated for disk matches - is this intentional?

Reply:
This might be leftover code. I ran a small test without `isMatch` and it still runs fine. If the full CI fails, I can put it back.

Reply:
I remember the issue now. There was something just in the disks where a later FM would sometimes overwrite the correct one. This `isMatch` variable prevented that.

Reply:
Is there something different in disks? I would not think so?! Can you explain what the issue was?

Reply:
There is a subtle difference here
https://github.com/cms-L1TK/firmware-hls/blob/6967d945656f2f53e94880590dce0a7e3427c1a5/TrackletAlgorithm/MatchProcessor.h#L1100-L1101
For the barrel layers we check `best_delta_phi` but for the disks we check `best_delta_rphi`. I found that sometimes a later stub after a correct match will still pass this criteria, so specifically checking `abs_delta_phi < best_delta_phi` (only after a match is already found) prevents this. In principle we can add this to the barrel layers as well to keep the code as similar as possible, but it will be redundant.

Reply:
Thanks for explaining - in the future we should review the code and see if we can simplify.